client/state/support/reducer.js

```diff
@@ -3,19 +3,42 @@
 /**
  * External dependencies
  */
+import debugFactory from 'debug';
 
+/**
+ * Internal dependencies
+ */
 import { combineReducers } from 'state/utils';
-import { SUPPORT_USER_ACTIVATE } from 'state/action-types';
+import { SUPPORT_SESSION_TRANSITION } from 'state/action-types';
+
+const debug = debugFactory( 'calypso:state:support:actions' );
+
+export const SESSION_NONE = 'none';
+export const SESSION_ACTIVE = 'active';
+export const SESSION_EXPIRED = 'expired';
 
-export function isSupportUser( state = false, { type } ) {
+export const SUPPORT_SESSION_STATE = {
+	NONE: 'none',
+	ACTIVE: 'active',
+	EXPIRED: 'expired',
+};
+
+function supportSession( state = SESSION_NONE, { type, nextState } ) {
 	switch ( type ) {
-		case SUPPORT_USER_ACTIVATE:
-			return true;
-	}
+		case SUPPORT_SESSION_TRANSITION:
+			if (
+				( state === SESSION_NONE && nextState === SESSION_ACTIVE ) ||
+				( state === SESSION_ACTIVE && nextState === SESSION_EXPIRED )
+			) {
+				return nextState;
+			}
 
-	return state;
+			debug( `invalid support session transition from '${ state }' to '${ nextState }'` );
+			return state;
+
+		default:
+			return state;
+	}
 }
 
-export default combineReducers( {
-	isSupportUser,
-} );
+export default combineReducers( { supportSession } );
```

Comment:
looks like this was already in a similar…_state_… but wrapping in `combineReducers()` feels redundant. we're going to end up with something like `state.supportSession.supporSession`

how about removing the wrapper and letting it be `export default supportSession`?

it may feel wrong because we recognize that the default export has no name but this gets imported in `state/index` and given a name there and in that big `combineReducers()` call for initial state.

```js
import { default as supportSession } from 'state/support/reducer';
```

Reply:
Good point, one of Brandon's commits removes the extra indirection.

Eventually, this flag could become part of `state.currentUser`. They are more like two parts of the same thing rather than two distinct thing.